hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorPartition.java

```diff
@@ -73,6 +81,23 @@ public Operation prepareReplicationOperation(int replicaIndex, boolean migration
         return new ReplicationOperation(map);
     }
 
+    public Map<String, Collection<ScheduledTaskDescriptor>> prepareOwnedSnapshot() {
+        Map<String, Collection<ScheduledTaskDescriptor>> containers = new HashMap<String, Collection<ScheduledTaskDescriptor>>();
+
+        if (logger.isFinestEnabled()) {
+            logger.finest("[Partition: " + partitionId + "] Prepare snapshot of partition owned tasks.");
+        }
+
+        for (ScheduledExecutorContainer container : getContainers()) {
+            SplitBrainMergePolicy mergePolicy = getMergePolicy(container.getName());
+            if (!(mergePolicy instanceof DiscardMergePolicy)) {
+                containers.put(container.getName(), container.prepareForReplication(true).values());
+            }
```

Comment:
The tasks should be canceled here - or in a `clear()` method like the one I've mentioned - https://github.com/hazelcast/hazelcast/pull/12046/files#r165355777